rclcpp/src/rclcpp/node.cpp

```diff
@@ -235,6 +237,7 @@ Node::Node(
   sub_namespace_(""),
   effective_namespace_(create_effective_namespace(this->get_namespace(), sub_namespace_))
 {
+  node_time_source_->attachClock(node_clock_);
```

Comment:
this actually tries to insert the same element `node_cloeck_` if `TimeSource` is not specified in `NodeOption`, but that is just fine since `std::unordered_set` checks the hash to avoid duplication.

Reply:
Would you prefer to live as it is, maybe with a comment, or make some kind of check to avoid this tentative re-insertion instead?